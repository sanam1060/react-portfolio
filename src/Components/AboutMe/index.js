import profileImage from "../../assets/images/profile-photo.jpg";

function AboutMe() {
  return (
    <section className="container d-flex flex-column align-items-center">
      <img
        data-testid="profilePhoto"
        className="hero-img my-3"
        src={profileImage}
        alt="Sanam Ganjian"
      />
      <p data-testid="profilePhoto" className="fs-1 fw-semibold mb-0">
        Sanam Ganjian
      </p>
      <h6 data-testid="profileUserTitle" className="fw-semibold">
        Recently Graduated Full Stack Web Developer
      </h6>
      <p data-testid="profileUserDescription">
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
        tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
        veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
        commodo consequat. Duis aute irure dolor in reprehenderit in voluptate
        velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint
        occaecat cupidatat non proident, sunt in culpa qui officia deserunt
        mollit anim id est laborum.
      </p>
    </section>
  );
}

export default AboutMe;
