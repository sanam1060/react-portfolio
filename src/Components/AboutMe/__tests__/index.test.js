import { render, cleanup, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import AboutMe from "..";

afterEach(cleanup);

describe("About Me component", () => {
  it("renders", () => {
    render(<AboutMe />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<AboutMe />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("contains image, name, title and describtion", () => {
    render(<AboutMe />);
    expect(screen.getByTestId("profilePhoto")).toHaveAttribute(
      "src",
      "profile-photo.jpg"
    );
    expect(screen.getByTestId("profileUserName")).toHaveTextContent(
      "Sanam Ganjian"
    );
    expect(screen.getByTestId("profileUserTitle")).toHaveTextContent(
      "Recently Graduated Full Stack Web Developer"
    );
    expect(screen.getByTestId("profileUserDescription")).toHaveTextContent(
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
    );
  });
});
