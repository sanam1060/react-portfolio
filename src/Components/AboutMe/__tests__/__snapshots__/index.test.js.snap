exports[`About Me component matches snapshot 1`] = `
<DocumentFragment>
  <section class="container d-flex flex-column align-items-center">
    <img alt="Sanam Ganjian" class="hero-img my-3" data-testid="profilePhoto" src="profile-photo.jpg"/>
    <p
      class="fs-1 fw-semibold mb-0"
      data-testid="profileUserName"
    >
      Sanam Ganjian
    </p>
    <h6
      class="fw-semibold"
      data-testid="profileUserTitle"
    >
      Recently Graduated Stack Web Developer
    </h6>
    <p data-testid="profileUserDescription">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
        tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
        veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
        commodo consequat. Duis aute irure dolor in reprehenderit in voluptate
        velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint
        occaecat cupidatat non proident, sunt in culpa qui officia deserunt
        mollit anim id est laborum.
    </p>
  </section>
</DocumentFragment>
`;